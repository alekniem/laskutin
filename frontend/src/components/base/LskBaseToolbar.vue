<template>
    <v-toolbar flat>
        <v-toolbar-title :id="titleId">
            <slot />
        </v-toolbar-title>
        <v-spacer />
        <slot name="button" />
    </v-toolbar>
</template>

<script>
export default {
    name: "LskBaseToolbar",

    props: {
        titleId: { type: String, required: true },
    },
};
</script>
