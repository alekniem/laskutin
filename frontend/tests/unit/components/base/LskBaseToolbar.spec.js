import LskBaseToolbar from "@/components/base/LskBaseToolbar";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseToolbar", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseToolbar, {
            propsData: {
                titleId: "TEST_TITLE_ID",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("no content in any slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseToolbar, {
            propsData: {
                titleId: "TEST_TITLE_ID",
            },
            slots: {
                default: "<div>DEFAULT</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in default slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseToolbar, {
            propsData: {
                titleId: "TEST_TITLE_ID",
            },
            slots: {
                button: "<div>BUTTON</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in button slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseToolbar, {
            propsData: {
                titleId: "TEST_TITLE_ID",
            },
            slots: {
                default: "<div>DEFAULT</div>",
                button: "<div>BUTTON</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in default slot and in button slot");
    });
});
