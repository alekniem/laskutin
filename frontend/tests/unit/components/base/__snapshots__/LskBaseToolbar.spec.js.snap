// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseToolbar should match snapshot: content in button slot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat"
        style="height: 64px;"
>
  <div class="v-toolbar__content"
       style="height: 64px;"
  >
    <div id="TEST_TITLE_ID"
         class="v-toolbar__title"
    >
    </div>
    <div class="spacer">
    </div>
    <div>
      BUTTON
    </div>
  </div>
</header>
`;

exports[`LskBaseToolbar should match snapshot: content in default slot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat"
        style="height: 64px;"
>
  <div class="v-toolbar__content"
       style="height: 64px;"
  >
    <div id="TEST_TITLE_ID"
         class="v-toolbar__title"
    >
      <div>
        DEFAULT
      </div>
    </div>
    <div class="spacer">
    </div>
  </div>
</header>
`;

exports[`LskBaseToolbar should match snapshot: content in default slot and in button slot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat"
        style="height: 64px;"
>
  <div class="v-toolbar__content"
       style="height: 64px;"
  >
    <div id="TEST_TITLE_ID"
         class="v-toolbar__title"
    >
      <div>
        DEFAULT
      </div>
    </div>
    <div class="spacer">
    </div>
    <div>
      BUTTON
    </div>
  </div>
</header>
`;

exports[`LskBaseToolbar should match snapshot: no content in any slot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat"
        style="height: 64px;"
>
  <div class="v-toolbar__content"
       style="height: 64px;"
  >
    <div id="TEST_TITLE_ID"
         class="v-toolbar__title"
    >
    </div>
    <div class="spacer">
    </div>
  </div>
</header>
`;
